README.md

```diff
@@ -131,6 +131,115 @@ fn main() {
 }
 ```
 
+## Example: TLS + Sync
+
+Instead of using a `TcpStream`, a `TlsStream<TcpStream>` will be used instead
+when the `tls` feature of `tarpc` is enabled. When using TLS, some additional
+information is required from the user. You will need to be make the
+`TlsAcceptor` and `TlsClientContext` structs. The `TlsAcceptor` and
+`TlsConnector` (which is required in `TlsClientContext`) structs are defined in
+the [native-tls](https://github.com/sfackler/rust-native-tls) crate which is
+exposed by `tarpc`.
+
+```rust
+#![feature(conservative_impl_trait, plugin)]
+#![plugin(tarpc_plugins)]
+
+#[macro_use]
+extern crate tarpc;
+
+use tarpc::sync::Connect;
+use tarpc::util::Never;
+use tarpc::TlsClientContext;
+use tarpc::native_tls::{Pkcs12, TlsAcceptor, TlsConnector};
+
+service! {
+    rpc hello(name: String) -> String;
+}
+
+#[derive(Clone)]
+struct HelloServer;
+
+impl SyncService for HelloServer {
+    fn hello(&mut self, name: String) -> Result<String, Never> {
+        Ok(format!("Hello, {}!", name))
+    }
+}
+
+fn tls_context() -> (TlsAcceptor, TlsClientContext) {
+     let buf = include_bytes!("test/identity.p12");
+     let pkcs12 = Pkcs12::from_der(buf, "password").unwrap();
+     let acceptor = TlsAcceptor::builder(pkcs12).unwrap().build().unwrap();
+     let client_cx = TlsClientContext::try_new("foobar.com").unwrap();
+
+     (acceptor, client_cx)
+}
+
+fn main() {
+    let addr = "localhost:10000";
+    let (acceptor, client_cx) = tls_context();
+    let _server = HelloServer.listen(addr, acceptor);
```

Comment:
Yep, you're right. It doesn't allow mixing TLS and non-TLS currently. That's what I was asking about on gitter before. However, I do think it's good to try to allow this.

I really like the idea of the `Config` structs for client and server. That change should then unify the API then across both stream types and I believe then we'd only need one version of the macro.

Let me give this a shot and see where I end up. Thinking about the client, I believe we'll need to add another generic parameter to indicate the stream type...